### ISSUE TITLE
Possible compatibility problem with iptables on Rocky Linux 9

### ISSUE THREAD

**Issue by rmatzinger** [webmin/webmin#2264]
Using Webmin to add and/or modify certain firewall rules in iptables on Rocky Linux 9 generates error messages like this:

  **WARNING! Your current IPtables configuration is invalid : ip6tables-restore v1.8.10 (nf_tables): unknown option “–state” 
  Error occurred at line: 11**

Trying to change this rule from ACCEPT to DROP causes the above error message: 
   **-A INPUT -p tcp -m tcp -m conntrack --dport 22 --state NEW --ctstate NEW -j ACCEPT**

Apparently Webmin is not using the "**--ctstate NEW**" syntax and instead is just using **--state**.
I don't know enough about iptables syntax to know what "**--ctstate**"  is.  Is it just new syntax for "**--state**"?

Webmin reports that these are the iptables package versions on my server:

iptables-libs 1.8.10-2.el9            | libxtables and iptables extensions userspace support
iptables-nft 1.8.10-2.el9             | nftables compatibility for iptables, arptables and ebtables
iptables-services 1.8.10-2.2.el9  | iptables and ip6tables services for iptables

Is this a bug or simply that Webmin's IPTables code isn't compatible with a newer verison of IPTables?

Anything I can do to resolve this?

Thanks.

--Richard

**Comment by jcameron** [comment]
This could be an issue with detecting the kernel version, which we use to determine whether to use `--state` or `--cstate`

What does `uname -r` output on your system?

**Comment by rmatzinger** [comment]
It is Rocky Linux 9.4 and I believe everything is up-to-date.
$ uname -r
5.14.0-427.28.1.el9_4.x86_64

**Comment by jcameron** [comment]
Ok, that looks fine.

Are you using the IPv4 or IPv6 firewall?

**Comment by rmatzinger** [comment]
Using the Linux IPv6 Firewall editor.   FirewallD is not running. 

Right now the Webmin display of the rule is: 
   **Accept | If protocol is TCP and destination port is 22 and state of connection is NEW and state of connection is NEW**
and that causes the "unknown option --state"  error message to be displayed.   Even when setting "Connection states" to Ignored, that error message is still displayed.

**Comment by jcameron** [comment]
So I just tested this on a fully updated Rocky 9 system, but wasn't able to reproduce the issue - the correct `--cstate` flag was used by Webmin.

If you delete the problem rule and re-create it, does the same issue still happen?

**Comment by rmatzinger** [comment]
Sorry, never occurred to me to delete and add it back.  Yes, that eliminated the error message.   As this is a relatively new server and my first with Rocky Linux, I had never looked at the IPv6 Iptables firewall before stumbling on this error when I modified that rule.   So I don't know if those rules were included in the Rocky installation, or if installing Webmin caused them to be added.  Somewhere I got the impression that Webmin has scripts to add some default firewall rules.  

Thank you for investigating this for me.  I appreciate that it is a big job to keep Webmin compatible with a bunch of Linux distributions.
